pkg/proxy/middleware.go

```diff
@@ -0,0 +1,77 @@
+package proxy
+
+import (
+	"bytes"
+	"log"
+	"math/rand"
+	"net/http"
+	"time"
+)
+
+var _ http.Handler = &RetryMiddleware{}
+
+type RetryMiddleware struct {
+	other      http.Handler
```

Comment:
`other` was a bit confusing to me at first. I don't have a suggestion either so feel free to ignore this comment.

Reply:
Good 👁️ . It is the next handler in the chain. I will give it a better name